tar/bsdtar.c

```diff
@@ -1001,19 +1002,28 @@ main(int argc, char **argv)
 
 	/* Attempt to load keyfile. */
 	if (bsdtar->keyfile != NULL) {
+		/*
```

Comment:
It seems that this would be better hoisted above this block, e.g.

```
if (bsdtar->keyfile && bsdtar->keyfile_from_config &&
    bsdtar->option_dryrun && access(bsdtar->keyfile, F_OK) {
        free(bsdtar->keyfile);
        bsdtar->keyfile = NULL;
        bsdtar->config_file_keyfile_failed = 1;
}
```

Reply:
(And that way it doesn't need the goto.)